apps/gitness/src/pages-v2/repo/repo-code.tsx

```diff
@@ -22,10 +22,18 @@ import { splitPathWithParents } from '../../utils/path-utils'
  */
 export const RepoCode = () => {
   const repoRef = useGetRepoRef()
-  const { spaceId, repoId, gitRef, resourcePath } = useParams<PathParams>()
-  const subResourcePath = useParams()['*'] || ''
-  const repoPath = `/${spaceId}/repos/${repoId}/code/${gitRef}`
-  const fullResourcePath = subResourcePath ? resourcePath + '/' + subResourcePath : resourcePath
+  const { spaceId, repoId, gitRef } = useParams<PathParams>()
+  const subCodePath = useParams()['*'] || ''
+
+  // Split the subCodePath into parts to avoid redundant splitting
+  const [rawSubGitRef = '', rawResourcePath = ''] = subCodePath.split('~')
+
+  // Normalize values to remove slash if present
+  const subGitRef = rawSubGitRef.endsWith('/') ? rawSubGitRef.slice(0, -1) : rawSubGitRef
+  const fullGitRef = subGitRef ? gitRef + '/' + subGitRef : gitRef
+  const fullResourcePath = rawResourcePath.startsWith('/') ? rawResourcePath.slice(1) : rawResourcePath
+
+  const repoPath = `/${spaceId}/repos/${repoId}/code/${fullGitRef}`
```

Comment:
NIT: We have this same snippet in two places, can we isolate it to a separate function? May be in your next PR.

Reply:
yes, will also add Edit / New file based routing to it with that